src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTargetSpelling.java

```diff
@@ -18,36 +18,85 @@
 import java.util.ArrayList;
 import java.util.Iterator;
 
+import com.google.gwt.core.client.JsArray;
+import com.google.gwt.event.dom.client.*;
 import org.rstudio.core.client.CsvReader;
 import org.rstudio.core.client.CsvWriter;
 import org.rstudio.core.client.ResultCallback;
 import org.rstudio.core.client.widget.NullProgressIndicator;
-import org.rstudio.studio.client.common.spelling.SpellChecker;
+import org.rstudio.studio.client.common.spelling.TypoSpellChecker;
 import org.rstudio.studio.client.server.Void;
+import org.rstudio.studio.client.workbench.views.output.lint.LintManager;
+import org.rstudio.studio.client.workbench.views.output.lint.model.LintItem;
+import org.rstudio.studio.client.workbench.views.source.editors.text.ace.Position;
+import org.rstudio.studio.client.workbench.views.source.editors.text.ace.Range;
 import org.rstudio.studio.client.workbench.views.source.editors.text.spelling.CheckSpelling;
 import org.rstudio.studio.client.workbench.views.source.editors.text.spelling.InitialProgressDialog;
 import org.rstudio.studio.client.workbench.views.source.editors.text.spelling.SpellingDialog;
 import org.rstudio.studio.client.workbench.views.source.model.DocUpdateSentinel;
 
 import com.google.gwt.event.shared.HandlerRegistration;
 
-public class TextEditingTargetSpelling implements SpellChecker.Context
+@SuppressWarnings("Duplicates") // TODO REMOVE THIS
```

Comment:
It's okay to check in TODOs, but you should add (a) your name, and (b) a couple of sentences about what needs to be done and why.

Reply:
In this case I definitely want to do the TODO before commit so this is a good reminder! Some code was duplicated for temporary  ease of testing and IntelliJ is VERY aggressive about letting you know this so the annotation was just to quiet IntelliJ for a bit.